wheatley/rhythm/regression.py

```diff
@@ -214,6 +217,22 @@ def log_warning(message: str) -> None:
                 log_warning(f"{new_peal_speed} is not positive")
             else:
                 self._peal_speed = new_peal_speed
+                # Don't try to calculate the regression line if _blow_interval is 0 (because the lin
+                # alg will cause a DivisionByZeroError).
+                if self._blow_interval == 0:
+                    return
+                # Make sure that the `Peal Speed` control on Ringing Room can update the speed in
+                # real time - this change makes no difference to the CLI version, because
+                # `self._peal_speed` cannot be changed whilst Wheatley is running.  This is not a
+                # long term solution (famous last words ;P), but for the initial RR release, I think
+                # this should happen.
+                peal_speed_interval = peal_speed_to_blow_interval(self._peal_speed, self.stage)
+                current_bell_time = self.real_time_to_blow_time(real_time)
+                # We want to find the line which has gradient `peal_speed_interval` and intersects with
+                # our current line at the current time (so that Wheatley's perception of time doesn't
+                # simply jump).
+                self._blow_interval = peal_speed_interval
```

Comment:
Could we add an info logging statement it's changing peal speed?

Reply:
Oh yes good point.